src/modelfree/scheduling.py

```diff
@@ -44,8 +68,16 @@ def get_func(self, func_type):
         :param func_type: str, must be in self.allowed_func_types
         :return: callable which takes one optional argument - frac_remaining
         """
-        assert func_type in self.allowed_func_types
-        return functools.partial(self.get_val, func_type)
+        self._validate_func_type(func_type)
+        if self.func_dict.get(func_type) is None:
+            return None
+        else:
+            return functools.partial(self.get_val, func_type)
+
+    def set_annealer_shaping_env(self, func_type, env):
```

Comment:
Seems like you made the right call to keep annealer's a class and not a decorator...

Reply:
Is `shaping_env` always a `RewardShapingVecEnv`? Do we ever use anything other than the `get_log_buffer_data`? If not, can we perhaps pass in `shaping_env.get_log_buffer_data` directly and rename `shaping_env` to e.g. `get_log`? It's best not to expose more of the interface than is strictly needed.